src/state/slices/opportunitiesSlice/resolvers/thorchainsavers/utils.ts

```diff
@@ -390,3 +394,50 @@ export const makeDaysToBreakEven = ({
   const daysToBreakEven = BigNumber.max(daysToBreakEvenOrZero, 1).toFixed(0)
   return daysToBreakEven
 }
+
+export const getThorchainFromAddress = async ({
+  accountId,
+  assetId,
+  getPosition,
+  accountMetadata,
+  wallet,
+}: {
+  accountId: AccountId
+  assetId: AssetId
+  getPosition: typeof getThorchainLendingPosition | typeof getThorchainSaversPosition
+  accountMetadata: AccountMetadata
+  wallet: HDWallet
+}): Promise<string> => {
+  const { chainId } = fromAssetId(assetId)
+  if (!isUtxoChainId(chainId)) return Promise.resolve(fromAccountId(accountId).account)
+
+  try {
+    const position = await getPosition({
+      accountId,
+      assetId,
+    })
+    if (!position) throw new Error(`No position found for assetId: ${assetId}`)
+    const address: string = (() => {
+      if ('owner' in position) return position.owner
+      if ('asset_address' in position) return position.asset_address
+      // For type completeness - if we have a response, we *should* either have an `owner` or `asset_address` property
+      return ''
+    })()
+    return chainId === bchChainId ? `bitcoincash:${address}` : address
+  } catch {
+    const accountType = accountMetadata?.accountType
+    const bip44Params = accountMetadata?.bip44Params
+    // Technically any chain adapter, but is only used for UTXO ChainIds in this file, so effectively an UTXO adapter
+    const chainAdapter = getChainAdapterManager().get(
+      chainId,
+    ) as unknown as UtxoBaseAdapter<UtxoChainId>
```

Comment:
one of the few times we could just use the base chain adapter type if we want haha. could be extra explicit with throw on `undefined` over the `!`..
```suggestion
    const chainAdapter = getChainAdapterManager().get(chainId)!
```

Reply:
a5f5c3cf1f